client/ayon_usd/plugins/publish/integrate_usd_output_processor_remap_relative.py

```diff
@@ -0,0 +1,133 @@
+import os
+
+import pyblish.api
+
+from ayon_core.pipeline import OptionalPyblishPluginMixin
+from ayon_core.pipeline.publish.lib import get_instance_expected_output_path
+
+# Avoid USD imports turning into errors when running in a host that does
+# not support the USD libs.
+try:
+    from pxr import Sdf, UsdUtils
+    HAS_USD_LIBS = True
+except ImportError:
+    HAS_USD_LIBS = False
+
+
+RELATIVE_ANCHOR_PREFIXES = ("./", "../", ".\\", "..\\")
+
+
+def get_drive(path) -> str:
+    """Return disk drive from path"""
+    return os.path.splitdrive(path)[0]
+
+
+class USDOutputProcessorRemapToRelativePaths(pyblish.api.InstancePlugin,
+                                             OptionalPyblishPluginMixin):
+    """Remap all paths in a USD Layer to be relative to its published path"""
+
+    label = "Process USD files to use relative paths"
+    families = ["usd"]
+
+    # Run just before the Integrator
+    order = pyblish.api.IntegratorOrder - 0.01
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
+        # Skip instance if not marked for integration
+        if not instance.data.get("integrate", True):
+            return
+
+        # Some hosts may not have USD libs available but can publish USD data.
+        # For those we'll log a warning.
+        if not HAS_USD_LIBS:
+            self.log.warning(
+                "Unable to process USD files to relative paths because "
+                "`pxr` USD libraries could not be imported.")
+            return
+
+        # For each USD representation, process the file.
+        for representation in instance.data.get("representations", []):
+            representation: dict
+
+            if representation.get("name") != "usd":
+                continue
+
+            # Get expected publish path
+            published_path = get_instance_expected_output_path(
+                instance,
+                representation_name=representation["name"],
+                ext=representation.get("ext")
+            )
+            published_path_root = os.path.dirname(published_path)
+
+            # Process all files of the representation
+            staging_dir: str = representation.get(
+                "stagingDir", instance.data.get("stagingDir"))
+
+            # Process single file or sequence of the representation
+            if isinstance(representation["files"], str):
+                # Single file
+                fname: str = representation["files"]
+                path = os.path.join(staging_dir, fname)
+                self.process_usd(path, start=published_path_root)
+            else:
+                # Sequence
+                for fname in representation["files"]:
+                    path = os.path.join(staging_dir, fname)
+                    self.process_usd(path, start=published_path_root)
```

Comment:
lets keep the string access to dicts to a minimum it makes it long to read with no advantage.
and in both cases causes multiple dict accesses for no reason. 

there might be a point in renaming fnames to a nicer name just for stile points. 

```suggestion
            # Process single file or sequence of the representation
            fnames = representation["files"]
            if isinstance(fnames, str):
                # Single file
                fname: str = fnames
                path = os.path.join(staging_dir, fname)
                self.process_usd(path, start=published_path_root)
            else:
                # Sequence
                for fname in fnames:
                    path = os.path.join(staging_dir, fname)
                    self.process_usd(path, start=published_path_root)
                           
```

Reply:
I mostly did this because I realllllly wanted to avoid someone to think `fnames` is actually a list of filenames, because in your case `fnames` may be `str` which to me felt confusing as a variable. I really then want to destroy/delete that variable as soon as possible. :(

Would you still like me to refactor the code even though it suffers from that issue?